core/chain-configs/src/client_config.rs

```diff
@@ -37,6 +37,70 @@ pub const DEFAULT_STATE_SYNC_NUM_CONCURRENT_REQUESTS_ON_CATCHUP_EXTERNAL: u32 =
 /// before giving up and downloading it from external storage.
 pub const DEFAULT_EXTERNAL_STORAGE_FALLBACK_THRESHOLD: u64 = 3;
 
+/// Describes the expected behavior of the node regarding shard tracking.
+/// If the node is an active validator, it will also track the shards it is responsible for as a validator.
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum TrackedConfig {
+    /// Tracks no shards (light client).
+    NoShards,
+    /// Tracks all shards.
+    AllShards,
+    /// Tracks shards that are assigned to given validator account.
+    ShadowValidator(AccountId),
+    /// Rotate between these sets of tracked shards.
+    /// Used to simulate the behavior of chunk only producers without staking tokens.
+    Schedule(Vec<Vec<ShardId>>),
+    /// Tracks shards that contain one of the given account.
+    Accounts(Vec<AccountId>),
+}
+
+impl TrackedConfig {
+    pub fn new_empty() -> Self {
+        TrackedConfig::NoShards
+    }
+
+    pub fn tracks_all_shards(&self) -> bool {
+        matches!(self, TrackedConfig::AllShards)
+    }
+
+    pub fn tracks_any_account(&self) -> bool {
+        if let TrackedConfig::Accounts(accounts) = &self {
+            return !accounts.is_empty();
+        }
+        false
+    }
+
+    /// For backward compatibility, we support `tracked_shards`, `tracked_shard_schedule`,
+    /// `tracked_shadow_validator`, and `tracked_accounts` as separate configuration fields,
+    /// in that order of priority.
+    pub fn from_deprecated_config_values(
```

Comment:
nice! What's your plan regarding deprecating the old configs in favour of the new ones? Perhaps something like the following:
* release x - log a warning if old configs are present
* release x + 1 - panic if old configs are present
* release x + 2 - remove old configs

Reply:
In the new archival node release notes, I would encourage validators to use the new config and mention that the old one will be deprecated soon. I'll also update the default configs in S3.

As for the steps for the upcoming releases, what you described sounds good to me, although I haven’t thought much about it yet. I’ll add it to the tracking issue and discuss it with the team, probably after the new archival node release. The main motivation for this refactor is to prepare for introducing `TrackedConfig::Shards` for single shard tracking.